### PR TITLE
Add comment sorting

### PR DESCRIPTION
**Pull request by jonasdeluna**
# Description

Adds comment sorting based on recency and likes/dislikes

# Result

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

<table>
    <tr>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>
			
        </td>
        <td>
			
        </td>
    </tr>
</table>

# Testing

- [ ] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Resolves ... (either GitHub issue or Linear task)

### CODE REVIEW

File: app/components/Comments/CommentView.tsx [hunk 1]
```diff
@@ -27,23 +67,81 @@ const CommentView = (props: Props) => {
...
+            size={20}
+            onClick={() => {
+              setDisplaySorting(!displaySorting);
+            }}
+            className={styles.sortIcon}
+            iconNode={<ListFilter />}
+          />
+          {displaySorting && (
+            <div className={styles.sortList}>
+              <FilterSection title="">
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:122]: Remove this

File: app/components/Comments/CommentView.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+.iconWrapper {
+  margin-left: 0.25rem;
+}
+
+.sortList {
+  margin-left: 0.5rem;
+  min-width: 10rem;
+}
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.css:16]: Please use variables

File: app/components/Comments/CommentView.tsx [hunk 3]
```diff
@@ -17,6 +26,37 @@ type Props = {
...
+  {
+    label: 'Nyeste',
+    value: 'createdAt',
+  },
+  {
+    label: 'Eldste',
+    value: 'createdAtInv',
+  },
+  {
+    label: 'Likes',
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:39]: Norwegian maybe? Likerklikk?? 😆

**Review by ivarnakken — CHANGES_REQUESTED** [review]

File: app/components/Comments/CommentView.tsx (hunk 3 above)
Comment (jonasdeluna) [reply app/components/Comments/CommentView.tsx:39]: replaced by 'Reaksjoner'

File: app/components/Comments/CommentView.tsx (hunk 1 above)
Comment (jonasdeluna) [reply app/components/Comments/CommentView.tsx:122]: Replaced by a dropdown

File: app/components/Comments/CommentView.tsx [hunk 4]
```diff
@@ -1,11 +1,15 @@
-import { Flex, LoadingIndicator } from '@webkom/lego-bricks';
+import { Flex, Icon, LoadingIndicator } from '@webkom/lego-bricks';
+import { ListFilter } from 'lucide-react';
+import moment from 'moment';
+import { useEffect, useState, type CSSProperties } from 'react';
 import CommentForm from 'app/components/CommentForm';
 import { generateTreeStructure } from 'app/utils';
+import Dropdown from '../Dropdown';
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:7]: Absolute import please 😄 I understand that this was probably just an auto import though

File: app/components/Comments/CommentView.tsx [hunk 5]
```diff
@@ -27,23 +51,87 @@ const CommentView = (props: Props) => {
...
+
       <Flex column gap="var(--spacing-sm)">
         {!formDisabled && <CommentForm {...commentFormProps} />}
 
-        <LoadingIndicator loading={!comments}>
-          {comments && (
+        <LoadingIndicator loading={!sortedComments}>
+          {sortedComments && (
             <CommentTree
-              comments={newOnTop ? tree.reverse() : tree}
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:46]: Why is `newOnTop` removed? This is used in the bdb comments where the order is switched (newest first). I still think we need this, as it should indicate the "default" order of items

File: app/components/Comments/CommentView.tsx [hunk 6]
```diff
@@ -27,23 +51,87 @@ const CommentView = (props: Props) => {
...
+                      key={o.value}
+                      onClick={() => {
+                        setDisplaySorting(!displaySorting);
+                        setOrdering(o);
+                      }}
+                      className={styles.dropdownElement}
+                    >
+                      {o.label}
+                    </Dropdown.ListItem>
+                    <Dropdown.Divider />
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:119]: Unnecessary divider at the end?

File: app/components/Comments/CommentView.tsx [hunk 7]
```diff
@@ -27,23 +51,87 @@ const CommentView = (props: Props) => {
...
+      >
+        <Title displayTitle={displayTitle} />
+        {
+          <Dropdown
+            show={displaySorting}
+            toggle={() => setDisplaySorting(!displaySorting)}
+            contentClassName={styles.changelogDropdown}
+            triggerComponent={
+              <Icon
+                name="arrow-down-wide-narrow"
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:95]:
```suggestion
```
Not used

File: app/components/Comments/CommentView.tsx [hunk 8]
```diff
@@ -27,23 +51,87 @@ const CommentView = (props: Props) => {
...
+                  setDisplaySorting(!displaySorting);
+                }}
+                className={styles.sortIcon}
+                iconNode={<ListFilter />}
+              />
+            }
+          >
+            <Dropdown.List>
+              {orderingOptions.map((o: Option) => {
+                return (
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:107]:
```suggestion
              {orderingOptions.map((o: Option) => (
```
Implicit return?

File: app/components/Comments/CommentView.tsx [hunk 9]
```diff
@@ -27,23 +51,87 @@ const CommentView = (props: Props) => {
...
-    newOnTop = false,
     contentAuthors,
   } = props;
   const commentFormProps = {
     contentTarget,
   };
-  const tree = generateTreeStructure(comments);
+  const [ordering, setOrdering] = useState<Option>(orderingOptions[0]);
+  const [sortedComments, setSortedComments] = useState<Comment[]>(comments);
+  const [displaySorting, setDisplaySorting] = useState(false);
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.tsx:61]: Is this state needed? Shouldn't this logic be handled by the Dropdown component?

**Review by ivarnakken — CHANGES_REQUESTED** [review]
Looking much better! Just some last nitpicks 😅

File: app/components/Comments/CommentView.tsx [hunk 10]
```diff
@@ -27,23 +51,87 @@ const CommentView = (props: Props) => {
...
     <div style={style}>
-      <Title displayTitle={displayTitle} />
+      <Flex
+        flex-row
+        justifyContent="flex-start"
+        alignItems="center"
+        className={styles.headerContainer}
+      >
+        <Title displayTitle={displayTitle} />
+        {
```
[the remark this answers is not in the document]
Comment (ivarnakken) [reply app/components/Comments/CommentView.tsx:88]: Perhaps we only want to show the dropdown if there are more than 1 comment? Not sure what's the best UX here

File: app/components/Comments/CommentView.tsx (hunk 9 above)
Comment (jonasdeluna) [reply app/components/Comments/CommentView.tsx:61]: I don't think it is?

File: app/components/Comments/CommentView.css [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+  margin-bottom: var(--spacing-sm);
+}
+
+.sortIcon {
+  flex-direction: column;
+}
+
+.dropdownElement {
+  padding: var(--spacing-sm);
+  cursor: pointer;
```
[the remark this answers is not in the document]
Comment (jonasdeluna) [reply app/components/Comments/CommentView.css:11]: Fixed

File: app/components/Comments/CommentView.tsx (hunk 6 above)
Comment (jonasdeluna) [reply app/components/Comments/CommentView.tsx]: It looks wack without it I think

File: app/components/Comments/CommentView.tsx [hunk 12]
```diff
@@ -27,21 +51,87 @@ const CommentView = (props: Props) => {
...
-    newOnTop = false,
     contentAuthors,
+    newOnTop,
   } = props;
   const commentFormProps = {
     contentTarget,
   };
-  const tree = generateTreeStructure(comments);
+  const [ordering, setOrdering] = useState<Option>(orderingOptions[0]);
+  const [sortedComments, setSortedComments] = useState<Comment[]>(comments);
```
Comment (eikhr) [inline app/components/Comments/CommentView.tsx:61]: Don't use a useState for this, just calculate based on `ordering` in the component. I think best practice is to avoid useState if possible.

File: cypress/e2e/event_visit_spec.js [hunk 13]
```diff
@@ -74,7 +74,7 @@ describe('View event', () => {
     cy.focused().type('This is the top comment');
     cy.contains('button', 'Kommenter').should('be.visible').click();
 
-    cy.get(c('Comment__comment')).last().contains('This is the top comment');
+    cy.get(c('Comment__comment')).first().contains('This is the top comment');
```
Comment (eikhr) [inline cypress/e2e/event_visit_spec.js:77]: Why is this changed?

**Review by eikhr — APPROVED** [review]
Looks good!

File: app/components/Comments/CommentView.css [hunk 14]
```diff
@@ -0,0 +1,9 @@
+.headerContainer {
+  margin-bottom: var(--spacing-sm);
+}
+
+.dropdownElement {
+  padding: var(--spacing-sm);
+  cursor: pointer;
+  width: var(--spacing-ss);
```
Comment (ivarnakken) [inline app/components/Comments/CommentView.css:8]: I don't think this exists

File: app/components/Comments/CommentView.tsx (hunk 5 above)
Comment (ivarnakken) [reply app/components/Comments/CommentView.tsx:46]: Did you look into this? I think we should support default sorting so that it's newest first on bdb
